### PR TITLE
Consistency, Tests, SBT Setup, Headers, Minor Refactoring

### PR DESCRIPTION
**Pull request by cchighman**
@shivsood @arvindshmicrosoft @rajmera3 @shanyu 
For visibility

### CODE REVIEW

**Comment by cchighman** [comment]
Note, this is from my branch "cleanup", and contains the commits from the PR in master.  I separated them into different PRs and only created them separately so that we could get an early look at this changes while I work complete the requests from the AAD PR.  I'll merge the commits afterwards.

**Comment by cchighman** [comment]
Well, I can see it picked up my SBT changes in the github lol.

File: .gitignore [hunk 1]
```diff
@@ -1,350 +1,35 @@
-## Ignore Visual Studio temporary files, build results, and
```
Comment (cchighman) [inline .gitignore:1]: This was imported from Spark

File: .scalafmt.conf [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#
```
Comment (cchighman) [inline .scalafmt.conf:1]: Imported from Spark

File: build.sbt [hunk 3]
```diff
@@ -6,20 +6,57 @@ version := "1.0.0"
...
+lazy val testScalastyle = taskKey[Unit]("testScalastyle")
+testScalastyle := scalastyle.in(Test).toTask("").value
+(test in Test) := ((test in Test) dependsOn testScalastyle).value
+
+lazy val compileScalastyle = taskKey[Unit]("compileScalastyle")
+compileScalastyle := scalastyle.in(Compile).toTask("").value
+(compile in Compile) := ((compile in Compile) dependsOn compileScalastyle).value
+scalastyleFailOnError := true
+
+// sbt-header
```
Comment (cchighman) [inline build.sbt:54]: Plugin to generate header

File: build.sbt [hunk 4]
```diff
@@ -6,20 +6,57 @@ version := "1.0.0"
...
+// Checkstyle build dependencies
+(checkstyle in Compile) := (checkstyle in Compile).triggeredBy(compile in Compile).value
+checkstyleSeverityLevel := Some(CheckstyleSeverityLevel.Error)
+
+// Code Coverage
+coverageEnabled := true
+coverageMinimum := 80
+coverageFailOnMinimum := true
+
+// scalastyle build dependency
```
Comment (cchighman) [inline build.sbt:44]: Plugin to make scalastyle linting a build requirement

File: build.sbt [hunk 5]
```diff
@@ -6,20 +6,57 @@ version := "1.0.0"
...
+  "com.microsoft.sqlserver" % "mssql-jdbc" % "7.2.1.jre8"
 )
+
 scalacOptions := Seq("-unchecked", "-deprecation", "evicted")
 
 // Exclude scala-library from this fat jar. The scala library is already there in spark package.
 assemblyOption in assembly := (assemblyOption in assembly).value.copy(includeScala = false)
+
+
+// Checkstyle build dependencies
```
Comment (cchighman) [inline build.sbt:35]: Support for checkstyle compilation that Spark uses.

File: checkstyle-suppressions.xml [hunk 6]
```diff
@@ -0,0 +1,51 @@
+<!--
```
Comment (cchighman) [inline checkstyle-suppressions.xml:1]: Imported from Spark

File: checkstyle.xml [hunk 7]
```diff
@@ -0,0 +1,190 @@
+<!--
```
Comment (cchighman) [inline checkstyle.xml:1]: Imported from Spark

**Comment by shivsood** [comment]
@cchighman Thanks for this work. These are great changes to have. My only concerns is that this is a big PR across several areas. Such a big PR is difficult to review and get the right owners onto for a review. Lets have 1 PR have one feature together with tests and samples.

Can u please submit separate PRs for each of these changes?  I see 3 PRs here
- headers/cleanup, license additions
- build, linting and style check, 
- additional tests and refactoring

Let's start with "headers/cleanup, license additions" @rajmera3 @arvindshmicrosoft need your review for these changes.

**Comment by cchighman** [comment]
> @cchighman Thanks for this work. These are great changes to have. My only concerns is that this is a big PR across several areas. Such a big PR is difficult to review and get the right owners onto for a review. Lets have 1 PR have one feature together with tests and samples.
> 
> Can u please submit separate PRs for each of these changes? I see 3 PRs here
> 
> * headers/cleanup, license additions
> * build, linting and style check,
> * additional tests and refactoring
> 
> Let's start with "headers/cleanup, license additions" @rajmera3 @arvindshmicrosoft need your review for these changes.

Plan on getting this done within the next day or so.

**Pull request closed without merging**